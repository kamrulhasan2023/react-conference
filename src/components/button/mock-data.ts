import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  variant: 'outline',
};
