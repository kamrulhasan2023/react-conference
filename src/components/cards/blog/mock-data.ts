import { IBlogCard } from './interface';

export const blogCardProps: IBlogCard = {
  href: 'https://www.lemonhive.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
    lqip: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  },
  title: 'Headless Website Development',
  description:
    'Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.',
  tags: [
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Reactjs' },
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Nextjs' },
  ],
  isFeatured: false,
};
