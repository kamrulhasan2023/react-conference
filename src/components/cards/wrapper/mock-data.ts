import { ICardWrapper } from './interface';

export const cardWrapperProps: ICardWrapper = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
