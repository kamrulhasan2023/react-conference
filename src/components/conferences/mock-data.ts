import {IConferences} from "./interface";
export const conferencesProps:IConferences = {
    title : "Conferences",
    conferences: [
        {   
            id:1,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:2,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:3,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
          startDate: "02 September, 2023",
        },
        {
            id:4,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
           
        },
        {
            id:5,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        }
    ]

}