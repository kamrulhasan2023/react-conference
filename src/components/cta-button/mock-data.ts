import { ICtaButton } from './interface';

export const ctaButtonProps: ICtaButton = {
  label: 'Cta button',
  href: '/',
  colorScheme: 'primary',
  isOpenNewTab: false,
};
