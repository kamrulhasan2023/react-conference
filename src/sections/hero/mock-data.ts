import { IHero } from './interface';

export const heroProps: IHero = {
  title: 'Lemon Hive, We Build Lightning Fast Websites',
  description:
    'We make secure, flexible and fast headless (Jamstack) websites and apps that improve Technical SEO, increase visitor numbers & conversion rates. We build with Sanity, BigCommerce, Prismic, Strapi, Contentful, WordPress, and other headless content management systems. We create front-ends with Vue, Svelte, Remix and React utilising Next.js, Gatsby, Nuxt, React Native, and others.',
  buttons: {
    buttonLeft: {
      label: 'Server Side Example',
      href: '/',
    },
    buttonRight: {
      label: 'Client Side Example',
      href: 'client-side',
    },
  },
};
