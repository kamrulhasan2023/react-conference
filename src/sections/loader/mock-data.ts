import { ILoaderSection } from './interface';

export const loaderSectionProps: ILoaderSection = {
  level: 'section',
};
