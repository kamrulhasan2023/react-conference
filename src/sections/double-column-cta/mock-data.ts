import { IDoubleColumnCtaSection } from './interface';
import { sectionDescriptionProps } from '@components/section-heading/mock-data';

export const doubleColumnCtaSectionProps: IDoubleColumnCtaSection = {
  sectionHeading: {
    title:
      'Weâ€™d love to help start your headless journey or  to help optimize your existing solution.',
  },
  rightContent: sectionDescriptionProps,
  ctaButton: {
    label: 'Book a project discovery call',
    href: 'https://example.com/',
    isOpenNewTab: true,
  },
};
