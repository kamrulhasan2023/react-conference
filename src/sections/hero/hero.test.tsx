import { render } from '@testing-library/react';
import Hero from '.';
import { heroProps } from './mock-data';

it('renders hero correctly', () => {
  const { getAllByRole } = render(<Hero {...heroProps} />);
  const element = getAllByRole('presentation')[0];

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
