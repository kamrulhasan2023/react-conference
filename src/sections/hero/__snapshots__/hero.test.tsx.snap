// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hero correctly 1`] = `
<section
  class="section-padding-primary flex min-h-screen items-center bg-black"
  role="presentation"
>
  <div
    class="container"
    role="presentation"
  >
    <div
      class="max-w-[1200px]"
    >
      <div
        class="space-y-12"
      >
        <h2
          class="text-4xl text-white md:text-5xl"
        >
          Lemon Hive, We Build Lightning Fast Websites
        </h2>
        <p
          class="text-xl text-white"
        >
          We make secure, flexible and fast headless (Jamstack) websites and apps that improve Technical SEO, increase visitor numbers & conversion rates. We build with Sanity, BigCommerce, Prismic, Strapi, Contentful, WordPress, and other headless content management systems. We create front-ends with Vue, Svelte, Remix and React utilising Next.js, Gatsby, Nuxt, React Native, and others.
        </p>
        <div
          class="inline-flex flex-wrap items-center gap-4"
        >
          <a
            class="inline-flex focus:outline-none focus-visible:outline-none focus:shadow-none focus-visible:shadow-none"
            href="/"
            target="_self"
          >
            <span
              aria-label="Server Side Example"
              class="relative z-10 group/button overflow-hidden inline-flex items-center justify-center gap-3 cursor-pointer text-white hover:text-white leading-[1.1] button bg-primary-text-white hover-bg-success-text-white sm"
            >
              <span>
                Server Side Example
              </span>
              <span
                class="absolute z-[-1] inset-0 invisible opacity-0 group-hover/button:visible group-hover/button:opacity-100 button-bg"
              />
            </span>
          </a>
          <a
            class="inline-flex focus:outline-none focus-visible:outline-none focus:shadow-none focus-visible:shadow-none"
            href="client-side"
            target="_self"
          >
            <span
              aria-label="Client Side Example"
              class="relative z-10 group/button overflow-hidden inline-flex items-center justify-center gap-3 cursor-pointer text-white hover:text-white leading-[1.1] button bg-success-text-white hover-bg-primary-text-white sm"
            >
              <span>
                Client Side Example
              </span>
              <span
                class="absolute z-[-1] inset-0 invisible opacity-0 group-hover/button:visible group-hover/button:opacity-100 button-bg"
              />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
